src/cpp/fastdds/xtypes/type_representation/TypeObjectUtils.cpp

```diff
@@ -0,0 +1,2952 @@
+        const CommonAliasBody& common_alias_body)
+{
+    empty_flags_consistency(common_alias_body.related_flags());
+    type_identifier_consistency(common_alias_body.related_type());
+}
+
+void TypeObjectUtils::hashid_builtin_annotation_not_applied_consistency(
+        const eprosima::fastcdr::optional<AppliedBuiltinMemberAnnotations>& ann_builtin)
+{
+    if (ann_builtin.has_value() && ann_builtin.value().hash_id().has_value())
+    {
+        throw InvalidArgumentError("@hashid builtin annotation cannot be applied to alias declaration");
+    }
+}
+
+void TypeObjectUtils::complete_alias_body_consistency(
+        const CompleteAliasBody& complete_alias_body)
+{
+    common_alias_body_consistency(complete_alias_body.common());
+    if (complete_alias_body.ann_custom().has_value())
+    {
+        applied_annotation_seq_consistency(complete_alias_body.ann_custom().value());
+    }
+    hashid_builtin_annotation_not_applied_consistency(complete_alias_body.ann_builtin());
+}
+
+void TypeObjectUtils::complete_alias_header_consistency(
+        const CompleteAliasHeader& complete_alias_header)
+{
+    complete_type_detail_consistency(complete_alias_header.detail());
+}
+
+void TypeObjectUtils::complete_alias_type_consistency(
+        const CompleteAliasType& complete_alias_type)
+{
+    empty_flags_consistency(complete_alias_type.alias_flags());
+    complete_alias_header_consistency(complete_alias_type.header());
+    complete_alias_body_consistency(complete_alias_type.body());
+}
+
+void TypeObjectUtils::complete_element_detail_consistency(
+        const CompleteElementDetail& complete_element_detail)
+{
+    if (complete_element_detail.ann_custom().has_value())
+    {
+        applied_annotation_seq_consistency(complete_element_detail.ann_custom().value());
+    }
+    hashid_builtin_annotation_not_applied_consistency(complete_element_detail.ann_builtin());
+}
+
+void TypeObjectUtils::common_collection_element_consistency(
+        const CommonCollectionElement& common_collection_element)
+{
+    member_flag_consistency(common_collection_element.element_flags());
+    type_identifier_consistency(common_collection_element.type());
+}
+
+void TypeObjectUtils::complete_collection_element_consistency(
+        const CompleteCollectionElement& complete_collection_element)
+{
+    common_collection_element_consistency(complete_collection_element.common());
+    complete_element_detail_consistency(complete_collection_element.detail());
+}
+
+void TypeObjectUtils::common_collection_header_consistency(
+        const CommonCollectionHeader& common_collection_header)
+{
+    bound_consistency(common_collection_header.bound());
+}
+
+void TypeObjectUtils::complete_collection_header_consistency(
+        const CompleteCollectionHeader& complete_collection_header)
+{
+    common_collection_header_consistency(complete_collection_header.common());
+    if (complete_collection_header.detail().has_value())
```

Comment:
As far as I understand the IDL specification, I think that every collection is anonymous as it has to be defined as alias or as a member, so the name is not the collection name per se, but the member name or the alias name which is applied to the anonymous collection. But regarding the question, no, I do not think there is any manner to check that the collection is anonymous to check that this optional structure is not set. The only option is to add to the API a flag and I do not think it is necessary because the user can already not set this parameter if not required.